### ISSUE TITLE
🐛 Theme switcher has no visible hover states

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#11571]
## 🐛 Bug

If you're tabbing through the page there are no visible hover states to know where the focus is.  There's also no visible hover state when using the mouse.

## 🦋 Expected Behaviour

We typically use a dark blue colour border over change the colour of a component to black on dark blue when hovering.

## 📸 Screenshot

![image](https://github.com/user-attachments/assets/0f22491a-7c60-48c5-b506-be65e6e9b8b1)

https://github.com/user-attachments/assets/c4097edd-7f0e-46bd-88ad-63e3e568ece9

## ✅ Acceptance Criteria

- [ ] When using the mouse, a hover state appears when mousing over the theme switcher
- [ ] When use the keyboard, a hover state appears when tabbing through the theme switcher

**Comment by petertgiles** [comment]
@JoshBeveridge Do you have a design suggestions for this one?

**Comment by JoshBeveridge** [comment]
@petertgiles When you say "hover" state, are you referring to the `focus-visible` styling when the user tab-focuses each button in the switcher?

Assuming that's the case:
![image](https://github.com/user-attachments/assets/fc5b57fc-8900-4e38-9c2a-d98a8c0d42ab)

**Comment by petertgiles** [comment]
@JoshBeveridge Yes, sorry, the tab-focus state.  When tabbing, the component as a whole is a single stop which is interactable with the arrow keys when focused.  The three individual buttons are not tab-focusable on their own, like in your example.

**Comment by tristan-orourke** [comment]
Todo: 
- keep the theme switcher as a single tab-focusable element
- some visual indicator that the element is focused
- blue focus colour on individual states should supersede the active yellow

**Comment by esizer** [comment]
Made an attempt to fix this but couldn't figure out how to do it. We have some pretty intense selectors happening for this component. I tried multiple different combinations and could not get it to work on all themes/modes :cry: 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c78f46a7016e31c0c8d97e13c5adc3c2cc813530/packages/ui/src/components/ToggleGroup/ToggleGroup.tsx#L41-L57
